app/test/unit/path-tests.ts

```diff
@@ -0,0 +1,24 @@
+import { expect } from 'chai'
+
+import { encodePathAsUrl } from '../../src/lib/path'
+
+describe('path', () => {
+  describe('encodePathAsUrl', () => {
+    if (__WIN32__) {
+      it('normalizes path separators on Windows', () => {
+        const dirName =
+          'C:/Users/shiftkey\\AppData\\Local\\GitHubDesktop\\app-1.0.4\\resources\\app'
+        const uri = encodePathAsUrl(dirName, 'folder/file.html')
+        expect(uri.startsWith('file:///C:/Users/shiftkey/'))
+      })
+    }
+
+    it('should encode hash symbol', () => {
+      const dirName =
+        'C:/Users/The Kong #2\\AppData\\Local\\GitHubDesktop\\app-1.0.4\\resources\\app'
+      const uri = encodePathAsUrl(dirName, 'index.html')
+      expect(uri.startsWith('file:///'))
+      expect(uri.indexOf('%23')).is.greaterThan(0)
```

Comment:
Could we, instead of checking that `%23` exists in the path, expect a constant value? I realize this means we'll have to conditionalize it for Windows and POSIX but I think it'd be much easier to read.